src/parse-stream.js

```diff
@@ -222,18 +239,17 @@ export default class ParseStream extends Stream {
         this.trigger('data', event);
         return;
       }
-      match = (/^#EXT-X-BYTERANGE:?([0-9.]*)?@?([0-9.]*)?/).exec(newLine);
+      match = (/^#EXT-X-BYTERANGE:?(.*)?$/).exec(newLine);
       if (match) {
         event = {
           type: 'tag',
           tagType: 'byterange'
         };
-        if (match[1]) {
-          event.length = parseInt(match[1], 10);
-        }
-        if (match[2]) {
-          event.offset = parseInt(match[2], 10);
-        }
+        const result = parseByterange(match[1]);
+
+        Object.keys(result).forEach(function(k) {
+          event[k] = result[k];
+        });
```

Comment:
What about setting `event` to be the response from `parseByterange` and then add `type` and `tagType` to that object?